src/twisted/logger/_format.py

```diff
@@ -136,6 +136,30 @@ def formatTime(when, timeFormat=timeFormatRFC3339, default=u"-"):
 
 
 
+def formatFailureTraceback(failure):
+    """
+    Format a failure traceback, assuming UTF-8 and using a replacement
+    strategy for errors.  Every effort is made to provide a usable
+    traceback, but should not that not be possible, a message and the
+    captured exception are logged.
+
+    @param failure: The failure to retrieve a traceback from.
+    @type failure: L{twisted.python.failure.Failure}
+
+    @return: The formatted traceback.
+    @rtype: L
+    """
+    try:
+        traceback = failure.getTraceback()
+        if isinstance(traceback, bytes):
+            traceback = traceback.decode('utf-8', errors='replace')
+    except Exception as e:
```

Comment:
I ended up refactoring this based on a review from @markrwilliams (https://github.com/twisted/twisted/pull/939#discussion_r179662072).  Originally I had bare except, specifically to catch everything, is there a reason `BaseException` is preferred?  To me the bare except more clearly communicates "everything", in part because it is normally avoided.  Just curious, I'll use `BaseException` in the update.

Reply:
Some linters complain about it, but more practically speaking: you can't `except as e:`.  So if you want to capture the variable you have to specify *some* kind of exception, and BaseException is the synonym for `except:`.  (I do wish this were spelled differently, like `except * as e` or something, but I'm not in charge of python syntax :))